src/components/SettingsView/RandomSeedInput.js

```diff
@@ -0,0 +1,52 @@
+import React, { useContext, useState } from 'react'
+import { string } from 'prop-types'
+import window from 'global/window'
+import TextField from '@material-ui/core/TextField'
+
+import FarmhandContext from '../Farmhand/Farmhand.context'
+
+import './RandomSeedInput.sass'
+
+export const RandomSeedInput = ({ search = window.location.search }) => {
+  const {
+    handlers: { handleRNGSeedChange },
+  } = useContext(FarmhandContext)
+
+  const [seed, setSeed] = useState(
+    new URLSearchParams(search).get('seed') ?? ''
+  )
+
+  /**
+   * @param {React.SyntheticEvent<HTMLInputElement>} e
+   */
+  const handleChange = e => {
+    setSeed(e.target.value)
+  }
+
+  /**
+   * @param {React.SyntheticEvent<HTMLFormElement>} e
+   */
+  const handleSubmit = e => {
+    e.preventDefault()
+
+    handleRNGSeedChange(seed)
+  }
+
+  return (
+    <form className="RandomSeedInput" onSubmit={handleSubmit}>
```

Comment:
pascal case className?! :sus:

Reply:
Hah that's the standard Create React App format for top-level component CSS classes. 🙂

https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/cra-template/template/src/App.js#L6-L7

Reply:
huh, i don't think i knew that. i've been doing all css modules work for the last few years now, but always did snake case before that. today i learned